### PR TITLE
Use Hetzner instead of Digitalocean for e2e tests

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
Digitalocean can be slow and flaky. Hetzner is also not the best hoster, but it's fast and super cheap.

* Hetzner's CX21 instance is €0.010 / h or roughtly €6 per month.
* DO's s-2vcpu-2gb droplets are $0.02232 / h or roughly $15 per month.

So we would save roughly 50%. Of course machines in these tests are super duper short lived (usually less than a minute), so price isn't really the driving factor here. Reliability and speed are.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: hack/ci/run-etcd-launcher-tests.sh [hunk 1]
```diff
@@ -34,18 +34,18 @@ export KIND_CLUSTER_NAME="${SEED_NAME:-kubermatic}"
...
+  name: e2e-hetzner
   namespace: kubermatic
 spec:
-  digitalocean:
-    token: ${DO_E2E_TESTS_TOKEN}
+  hetzner:
+    token: ${HZ_E2E_TOKEN}
 EOF
-retry 2 kubectl apply -f preset-digitalocean.yaml
+retry 2 kubectl apply -f preset-hetzner.yaml
```
Comment (embik) [inline hack/ci/run-etcd-launcher-tests.sh:48]: Why don't we use the `kubeadm` ("BYOC") provider for this (and perhaps other tests that only concern the control plane)? That makes sure we do not rely on a cloud API to be up and running when not needed, as we only get the control plane. But that would suffice for `etcd-launcher` tests.

Comment (xrstf) [reply hack/ci/run-etcd-launcher-tests.sh:48]: Except for one test, all of the tests use 0 nodes. But for one e2e test, we do want an actual node with an actual cloud provider (or kind, maybe).

Comment (embik) [reply hack/ci/run-etcd-launcher-tests.sh:48]: Yeah, I get that, I don't mean to suggest that we completely drop using a cloud provider like Hetzner, but how do you think about using `byoc` where it makes sense (like the etcd-launcher test above)?

Comment (xrstf) [reply hack/ci/run-etcd-launcher-tests.sh:48]: We only have one single function to create a cluster in our tests, and that one uses Hetzner. I did not want to duplicate code needlessly.

Comment (embik) [reply hack/ci/run-etcd-launcher-tests.sh:48]: Okay, sounds reasonable for now.

**Review by embik — APPROVED** [review]

**Comment by xrstf** [comment]
@floreks PTAL

**Pull request merged**